### PR TITLE
Create workflow branch

### LINKED ISSUE #37: Create release branch for workflows
See https://github.com/keycloak-rel-testing/keycloak-rel/issues/13

### PR DESCRIPTION
**Pull request by stianst**
Closes #37

### CODE REVIEW

**Comment by stianst** [comment]
Create branches test run:
https://github.com/keycloak-rel-testing/keycloak-rel/actions/runs/3135016639

**Comment by stianst** [comment]
Create branches test run:
https://github.com/keycloak-rel-testing/keycloak-rel/actions/runs/3135030709

Confirmed workflow branch was created:
image

**Comment by stianst** [comment]
Delete branch test run:
https://github.com/keycloak-rel-testing/keycloak-rel/actions/runs/3135040912

Confirmed branch was removed:
image

File: .github/workflows/release-nightly.yml [hunk 1]
```diff
@@ -12,7 +12,6 @@ jobs:
   show-inputs:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
```
Comment (stianst) [inline .github/workflows/release-nightly.yml:15]: Not relevant to this PR, but sneaking this small change in as we don't need to checkout the repo for this

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -19,7 +19,6 @@ jobs:
   show-inputs:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
```
Comment (stianst) [inline .github/workflows/release.yml:22]: Not relevant to this PR, but sneaking this small change in as we don't need to checkout the repo for this

File: .github/workflows/branch-create.yml [hunk 3]
```diff
@@ -15,11 +15,32 @@ defaults:
 
 jobs:
 
+  show-inputs:
+    runs-on: ubuntu-latest
+    steps:
+      - run: |
+          echo "Major.minor version: ${{ inputs.majorMinorVersion }}  " >> $GITHUB_STEP_SUMMARY
```
Comment (jonkoops) [inline .github/workflows/branch-create.yml:22]: Can be a single line.
```suggestion
      - run: echo "Major.minor version: ${{ inputs.majorMinorVersion }}  " >> $GITHUB_STEP_SUMMARY
```

File: .github/workflows/branch-create.yml [hunk 4]
```diff
@@ -15,11 +15,32 @@ defaults:
...
+    name: Create workflow branch
+    runs-on: ubuntu-latest
+    needs: [env]
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          path: keycloak-rel
+          ref: 'main'
```
Comment (jonkoops) [inline .github/workflows/branch-create.yml:37]: No need for quotes here.
```suggestion
          ref: main
```

File: .github/workflows/branch-delete.yml [hunk 5]
```diff
@@ -18,14 +18,36 @@ defaults:
 
 jobs:
 
+  show-inputs:
+    runs-on: ubuntu-latest
+    steps:
+      - run: |
+          echo "Major.minor version: ${{ inputs.majorMinorVersion }}  " >> $GITHUB_STEP_SUMMARY
```
Comment (jonkoops) [inline .github/workflows/branch-delete.yml:25]:
```suggestion
      - run: echo "Major.minor version: ${{ inputs.majorMinorVersion }}  " >> $GITHUB_STEP_SUMMARY
```

File: .github/workflows/branch-delete.yml [hunk 6]
```diff
@@ -18,14 +18,36 @@ defaults:
...
+    runs-on: ubuntu-latest
+    needs: [env]
+    if: github.event.inputs.majorMinorVersion != 'main' && github.event.inputs.majorMinorVersion == github.event.inputs.majorMinorVersionConfirmation
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          path: keycloak-rel
+          ref: 'main'
```
Comment (jonkoops) [inline .github/workflows/branch-delete.yml:41]:
```suggestion
          ref: main
```

**Review by jonkoops — APPROVED** [review]

**Pull request merged**